source/titfront/src/components/PythonShell.tsx

```
@@ -0,0 +1,181 @@
+    }
+  };
+
+  const handleClick = (event: MouseEvent<HTMLElement>) => {
+    event.preventDefault();
+
+    // If anything is selected, don't focus the input. User will have to click
+    // to unselect, and we'll focus the input then.
+    const selection = window.getSelection();
+    if (selection && selection.toString().length > 0) return;
+
+    inputRef.current!.focus();
```

Biome (1.9.4)

[error] Forbidden non-null assertion.

Unsafe fix: Replace with optional chain operator ?. This operator includes runtime checks, so it is safer than the compile-only non-null assertion operator


(lint/style/noNonNullAssertion)